### DISCUSSION TITLE
Unable to delete replicated objects immediately in one-way bucket replication

### DISCUSSION THREAD

**Discussion by arijitdeltax** [minio/minio#16854] [Q&A]
### I unable to delete objects on replicated(secondary) bucket automatically  when I delete objects on the primary bucket.

I am currently using [minio docker](https://min.io/docs/minio/container/index.html)

`mc --version` returns the following.
```bash
mc version RELEASE.2023-01-28T20-29-38Z (commit-id=2e95a70c98fb9c2629cd89817b8759bfa109a4d0)
Runtime: go1.19.4 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

I am replicating the bucket one-way using the following command:
```bash
mc replicate add $miniouser/$primaryminiobucket --priority 0 --sync --remote-bucket https://$accessKey:$secretKey@$serverIP:$minioport/$secondaryminiobucket --replicate "delete,delete-marker,existing-objects,metadata-sync"
``` 

When I am uploading as data to the primary bucket using `mc cp --recursive --preserve`, it is replicated to the secondary bucket  almost immediately.

But when I am deleting objects from the primary server using `mc rm --recursive --force --versions`, the objects on the primary server are removed **but the objects on the secondary server are not removed** :(

**Comment by harshavardhana** [comment]
@poornas ^^

**Comment by kannappanr** [comment]
I tried to repro this on play server. 

```
mc replicate add play/src --priority 0 --sync --remote-bucket https://minioadmin:minioadmin@play.min.io/dest --replicate "delete,delete-marker,existing-objects,metadata-sync"
Replication configuration rule applied to play/src successfully.
```
```
mc ls --versions play/src
[2023-03-28 02:19:48 PDT] 1.4MiB STANDARD c2a53091-833e-4b8d-86b8-7134cc736bb6 v1 PUT test

mc ls --versions play/dest
[2023-03-28 02:19:48 PDT] 1.4MiB STANDARD c2a53091-833e-4b8d-86b8-7134cc736bb6 v1 PUT test

mc rm -vid c2a53091-833e-4b8d-86b8-7134cc736bb6 play/src/test
Removed `play/src/test` (versionId=c2a53091-833e-4b8d-86b8-7134cc736bb6).

mc ls --versions play/dest
```
```
mc ls --versions play/dest
[2023-03-28 02:23:07 PDT] 1.4MiB STANDARD b49f2962-26e6-4e7b-b0c5-657760e6b6e3 v1 PUT test

mc ls --versions play/src
[2023-03-28 02:23:07 PDT] 1.4MiB STANDARD b49f2962-26e6-4e7b-b0c5-657760e6b6e3 v1 PUT test

mc rm --recursive --force --versions play/src
Removed `play/src/test` (versionId=b49f2962-26e6-4e7b-b0c5-657760e6b6e3).

mc ls --versions play/dest
```

**Comment by poornas** [comment]
@arijitdeltax , which version of minio is this? This is not reproducible with latest

**Comment by arijitdeltax** [comment]
@kannappanr  @poornas  the problem arises when the replication takes place from one  to many minio servers. It works perfectly well in one to one replication. I tested it on replication from 1 primary to 10 secondary minio nodes. 

I'm using the latest tag for the minio container: 
```bash
$ docker run --rm quay.io/minio/minio --version
minio version RELEASE.2023-01-12T02-06-16Z (commit-id=7bc95c47a322971aff7d4d4c270dcf28a933e84b)                                              
Runtime: go1.19.4 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.
```

**Reply by poornas** [reply]
First of all, do you have the bandwidth to run replication to 10 targets? 

Tested against 4 targets which is itself quite unrealistic for production setups - even with one of the targets down, replication holds on to state and completes delete replication after the target is back.

Do you have any server logs on the primary? Please use latest release - I do not see any issue with replication behavior
